### PR TITLE
Add LinuxAdmin::NetworkInterface.list

### PR DESCRIPTION
**Pull request by agrare**
Previously there wasn't a method for listing interfaces on the system, which was okay because we assumed an eth0 was available but as we move away from that and use the standard predictable network interface names it would be helpful to get a list.

This can be followed up with attributes indicating if an interface is a LOOPBACK type and if the interface is the dev registered for the default route.

Related https://github.com/ManageIQ/manageiq-appliance-build/pull/573

### CODE REVIEW

File: lib/linux_admin/network_interface.rb [hunk 1]
```diff
@@ -20,6 +20,23 @@ def self.dist_class(clear_cache = false)
...
+        .map { |net| net["ifname"] }
+        .map { |iface| new(iface) }
+    rescue AwesomeSpawn::CommandResultError => e
+      raise NetworkInterfaceError.new(e.message, e.result)
+    end
+
+    private_class_method def self.ip_link
+      require "json"
+
+      ip_link = Common.run!(Common.cmd("ip"), :params => ["--json", "link"])
```
Comment (agrare) [inline lib/linux_admin/network_interface.rb:34]: NOTE I think the `--json` output format would be helpful for e.g. `ip addr show` also which currently does some regex parsing of the output but I wanted to leave that for a standalone PR

**Comment by Fryguy** [comment]
`ip` doesn't exist on a Mac - is that ok for this PR?

**Comment by agrare** [comment]
> ip doesn't exist on a Mac - is that ok for this PR?

I think you're looking for the `OsxAdmin` repo :laughing: 

In all seriousness though, `LinuxAdmin::NetworkInterface#initialize` calls `ip addr show @interface`, and `ip route` already so I don't think this would work on a mac as is.

**Comment by Fryguy** [comment]
code-wise LGTM - There are a lot of rubocop issues, and I wasn't sure if the new code was just following an existing pattern, or if this should be cleaned up.

**Comment by agrare** [comment]
> There are a lot of rubocop issues, and I wasn't sure if the new code was just following an existing pattern

Yeah that is just how the specs are currently written so I wanted to keep it looking consistent.

File: lib/linux_admin/network_interface.rb (hunk 1 above)
Comment (agrare) [reply lib/linux_admin/network_interface.rb]: https://github.com/ManageIQ/linux_admin/pull/251

**Comment by agrare** [comment]
@Fryguy I tackled the rubocop warnings here in https://github.com/ManageIQ/linux_admin/pull/252

**Comment by Fryguy** [comment]
Merged #252 252, but now this is conflict

**Comment by agrare** [comment]
Yes this added some ip command mocks so conflict was expected

**Comment by agrare** [comment]
Rebased

**Review by Fryguy — APPROVED** [review]
